site/docs/concepts/derivations.md

```diff
@@ -7,700 +7,769 @@ import Mermaid from '@theme/Mermaid';
+Flow is responsible for the persistence and replication of your SQLite database,
+and the SQLite connector tracks and will apply your migrations as needed.
 
-<Tabs>
-<TabItem value="acmeBank.flow.yaml" default>
+### Performance
+
+Your observed performance will *of course* depend on the specifics of your use case,
+including the size of your task states
+and the complexity of your source documents and transformations.
+
+Generally speaking, SQLite is very performant
+and Flow's SQLite connector strives to drive it as efficiently as possible.
+Real-world use cases are observed to process many **tens
+of thousands** of documents per second on a single core.
+
+Flow can also scale your task without downtime
+by creating point-in-time clones of the database
+that subdivide the overall workload and storage of the task.
+Once created, these subdivisions process in parallel across
+multiple physical machines to enhance performance.
+
+## TypeScript
+
+Flow's TypeScript derivation connector transforms your source documents
+by executing methods of a TypeScript class which you implement.
+TypeScript derivations are executed using [Deno](https://deno.land/)
+and let you take advantage of the broad ecosystem of
+available third-party JavaScript and TypeScript libraries,
+as well as native code compiled to WASM.
+
+TypeScript derivations are strongly typed:
+Flow maps the JSON schemas of your source and output collections
+into corresponding TypeScript types,
+which are type-checked as you develop and test your derivation.
+This helps catch a wide variety of potential bugs and avoid accidental violations of your collection data contracts.
+
+### Modules
+
+The bulk of a TypeScript derivation lives in its associated module,
+which is a TypeScript source file that exports the class that implements your derivation.
+
+Each derivation also has an accompanying, generated interfaces module.
+Interface modules are managed by Flow: any changes you make will be ignored.
+The `flowctl generate --source path/to/my/derivation.flow.yaml` CLI command
+will generate interface modules under paths like
+`flow_generated/typescript/acmeCo/my-derivation.ts`,
+under the top-level directory under `--source` having a `flow.yaml` or `flow.json` file.
+
+It will also generate a `deno.json` file in your top-level directory,
+which is designed to work with developer tooling like
+[VSCode's Deno extension](https://marketplace.visualstudio.com/items?itemName=denoland.vscode-deno).
+
+[See the Current Account Balances tutorial for a concrete example of modules](#current-account-balances).
+
+### State
+
+The abstract `IDerivation` class generated within the interfaces module
+includes additional, experimental methods which can be used for
+persisting and recovering internal state of the connector.
+
+Consult the generated implementation and feel free to reach out to support
+if you'd like more information on building stateful TypeScript derivations.
+
+## Specification
```

Comment:
I think this section could be more effective *before* the **Connectors** heading. Then it’s available just after the tutorial as reference, and gives us a high-level overview before we go into detail in the next few sections.

Reply:
Done 👍 

I was unsure if this was a bit too reference-y, which is why I moved it down, but if you think it fits here then I'm definitely sold.